### ISSUE TITLE
Copying selection as PNG results in a flicker on all open windows 

### ISSUE THREAD

**Issue by shrayasr** [excalidraw/excalidraw#5778]
Hi folks, 

Upfront, thanks to the entire team for all the effort that is poured into making this tool. It has made everything easier and was partially responsible for saving our team a lot of troubles when we were forced to go remote during the pandemic. 

---

I'm here today because I seem to be facing a lot of issue in one regard. Before I get there, here is my workflow: 

1. Design something on excalidraw 
2. Select said design 
3. Copy to clipboard
4. Add to notion / send to engineer (via slack, teams, etc) 

I keep everything previously designed as well. So I have one HUGE canvas. 

The issue right now seems to be coming from step (3) in the workflow. Whenever I copy to clipboard as a PNG, all the open windows seem to be getting focused and losing focus. 

This is best illustrated with a gif: 

![4VPe5ANWbd](https://user-images.githubusercontent.com/240368/196894030-cfd6f93b-1eaf-45cd-a163-2fd72dd74a54.gif)

If you notice above, the instant I hit on the "copy to clipboard as PNG" button, both the windows start flickering. Once this happens, I can't type in any window or do any interaction because the focus just keeps going off (I attempted to write `notepad` in the run window by the side and only ended up registering `n` there)

I am able to mitigate this in 2 ways: 

1. Restarting the machine (which is my last resort, since I might have a lot of work being done then) 
2. Locking the machine. Doing so, and leaving it locked for a few seconds and then relogging in seems to fix this problem. 

---

I hope that I was able to convey the point across. Please do feel free to ask for more clarification as required. 

Thanks again for your efforts! Cheers ✨

**Comment by dwelle** [comment]
Hey @shrayasr, and thanks for your words!

As for the issue, I've not seen this yet. Can't repro myself (I'm on Win 10, you seem to be on Win 11 — /cc @maielo can you check on your end?).

That said, what you're describing is an OS issue, so there's not much we can do here. So the best hope here is to try to reproduce elsewhere, narrow down the issue (specific browser, OS update, ...), and report to the browser.

Also, I see you're using Brave. Does it also reproduce in Chrome or Edge?

**Comment by maielo** [comment]
@dwelle just testing it and can not reproduce it. Also was testing it with huge scene & brave.

**Comment by shrayasr** [comment]
> Also, I see you're using Brave. Does it also reproduce in Chrome or Edge?

Yes it happens on all browsers, @dwelle 

> That said, what you're describing is an OS issue, so there's not much we can do here. So the best hope we have is to try to reproduce elsewhere, narrow down the issue (specific browser/engine, OS update, ...), and report to the browser.

I was thinking that it was an OS issue too. And I believe you might be right. Today I copied a selection in photoshop and this happened for me even there. 

So I think its some kind of a windows 11, copying an image related issue then 🤔

**Comment by shrayasr** [comment]
> So I think its some kind of a windows 11, copying an image related issue

https://community.adobe.com/t5/photoshop-ecosystem-bugs/p-windows-11-icons-and-menus-quot-flicker-quot-after-copying-a-selection-in-photoshop/idc-p/12958792#M56952

Seems to be the perfect explanation for it right now. Copying an image to the clipboard causes this. When I copy something else (with great difficulty I might add), then the flicker stops. 🤦🏽

**Comment by shrayasr** [comment]
OK dug around a bit more and it looks like its some combo of WSL2, Docker and windows 11 that's causing this. 

Greenshot (the screenshot tool) also has an issue on this: https://github.com/greenshot/greenshot/issues/348 which they seem to have resolved by using a different way to copy to windows' clipboard. Recap here: https://github.com/greenshot/greenshot/issues/348#issuecomment-1033470769

But the core problem seems to be in WSL's GUI layer as I found via this issue here: https://github.com/microsoft/wslg/issues/443

I will try the fixes mention and get back. But as of now, this is **not** an excalidraw related issue. It seems like excalidraw was just in the line of fire, sorry. I hope we can keep this open until resolved from my end, so that it can be a reference for passerby like me.

**Comment by shrayasr** [comment]
Okkkk. This seems to be fixed now. Just adding the solution here for clarity: 

I tried to do a `wsl --update` as advised [here](https://github.com/microsoft/wslg/issues/443#issuecomment-1156128086) but this didn't fix it. Looks like I was on an older version of WSLg (the GUI layer) since as per their instruction `wsl --version` gave me an error. 

Then I installed WSL preview from here: https://aka.ms/wslstorepage and it seems to have been fixed now when copying an image from both excalidraw and photoshop 

My `wsl --version` now shows 

```
WSL version: 0.70.0.0
Kernel version: 5.15.68.1
WSLg version: 1.0.45
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22000.1098
```

Closing this. Thanks for your assistance folks! :)
